rest/rest-smartproxy/src/main/java/org/ow2/proactive_grid_cloud_portal/smartproxy/RestSmartProxyImpl.java

```diff
@@ -312,44 +303,66 @@ public void addEventListener(SchedulerEventListenerExtended listener, boolean my
+            String remotePath = extractRemotePath(userSpaces, inputSpaces);
 
-        if (!inputSpace.startsWith(userSpace)) {
-            // NOTE: only works for USERSPACE urls
-            logger.warn("RestSmartProxy does not support data transfers outside USERSPACE.");
-            return false;
-        }
-        String remotePath = inputSpace.substring(userSpace.length() + (userSpace.endsWith("/") ? 0 : 1));
-        String jname = job.getName();
-        logger.debug("Pushing files for job " + jname + " from " + localInputFolderPath + " to " + remotePath);
-        TaskFlowJob tfj = job;
-        for (Task t : tfj.getTasks()) {
-            logger.debug("Pushing files for task " + t.getName());
-            List<String> includes = Lists.newArrayList();
-            List<String> excludes = Lists.newArrayList();
-
-            List<InputSelector> inputFilesList = t.getInputFilesList();
-
-            if (inputFilesList != null) {
-                for (InputSelector is : inputFilesList) {
-                    addfileSelection(is.getInputFiles(), includes, excludes);
-                }
+            if (remotePath == null) {
+                // NOTE: only works for USERSPACE urls
+                throw new IllegalArgumentException("Could not extract remote path using inputSpace=" + inputSpaces +
+                                                   " and userSpace=" + userSpaces);
             }
+            String jname = job.getName();
+            logger.debug("Pushing files for job " + jname + " from " + localInputFolderPath + " to " + remotePath);
+            TaskFlowJob tfj = job;
+            for (Task t : tfj.getTasks()) {
+                logger.debug("Pushing files for task " + t.getName());
+                List<String> includes = Lists.newArrayList();
+                List<String> excludes = Lists.newArrayList();
+
+                List<InputSelector> inputFilesList = t.getInputFilesList();
+
+                if (inputFilesList != null) {
+                    for (InputSelector is : inputFilesList) {
+                        addfileSelection(is.getInputFiles(), includes, excludes);
+                    }
+                }
 
-            LocalDirSource source = new LocalDirSource(localInputFolderPath);
-            source.setIncludes(includes);
-            source.setExcludes(excludes);
+                LocalDirSource source = new LocalDirSource(localInputFolderPath);
+                source.setIncludes(includes);
+                source.setExcludes(excludes);
 
-            RemoteDestination dest = new RemoteDestination(USER, remotePath);
-            restDataSpaceClient.upload(source, dest);
+                RemoteDestination dest = new RemoteDestination(USER, remotePath);
+                restDataSpaceClient.upload(source, dest);
+            }
+            logger.debug("Finished push operation from " + localInputFolderPath + " to " + remotePath);
+
+        } catch (Throwable error) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)

Reply:
ok